### ISSUE TITLE
Rule proposal: prefer await instead of returning promises that resolves with void in async functions

### ISSUE THREAD

**Issue by jimmywarting** [typescript-eslint/typescript-eslint#4597]
### Description

Given this two functions:
```ts
async function init(): Promise<void> {
  await doSomeOtherLogic()
}

async function start(): Promise<void> {
  return init()
}
```
it makes it clear that `init` don't return anything. Therefore it makes since not to also return the `init()` call either and therefore it's best to use await instead to keep making sure that start also returns undefined.

### Fail

```ts
async function init(): Promise<void> {
  await doSomeOtherLogic()
}

async function start(): Promise<void> {
  return init()
}
```

### Pass

```ts
async function init(): Promise<void> {
  await doSomeOtherLogic()
}

async function start(): Promise<void> {
  await init()
}
```

**Comment by bradzacher** [comment]
I'm not sure I understand why it makes sense to await instead of return here? Why does the `void` make any difference?
